lib/jsonapi-serializers/serializer.rb

```diff
@@ -364,6 +364,15 @@ def self.find_recursive_relationships(root_object, root_inclusion_tree, results)
           is_collection = true
           object = serializer.has_many_relationships[unformatted_attr_name]
         end
+
+        if is_valid_attr && attribute_name.include?("_")
+          expected_name = serializer.format_name(attribute_name)
```

Comment:
I can certainly make this change, but wouldn't the existing be safe since we're also guarding for `is_valid_attr`, that is the unformatted name is a valid `has_one` or `has_many` relationship name?